backend/gateway/mux/mux.go

```diff
@@ -180,6 +184,15 @@ func customResponseForwarder(ctx context.Context, w http.ResponseWriter, resp pr
 	return nil
 }
 
+// forward all X- headers and add the grpcgateway prefix
+func customMatcher(key string) (string, bool) {
+	if strings.HasPrefix(key, xHeader) {
```

Comment:
http headers are not technically case sensitive so you need to match `X-` and `x-`.

or you can call the `textproto.CanonicalMIMEHeaderKey` first like the default matcher so they always end up in the same format with `X-`. i think that's the best option.

also add test case for this scenario.